frmts/vrt/vrtsources.cpp

```diff
@@ -3256,8 +3302,8 @@ CPLErr VRTComplexSource::RasterIOProcessNoData(
                 {
                     // Convert first to the VRTRasterBand data type
                     // to get its clamping, before outputting to buffer data type
-                    GDALCopyWords(&paSrcData[idxBuffer], eSourceType, 0,
-                                  abyTemp, eVRTBandDataType, 0, 1);
+                    CopyWord(&paSrcData[idxBuffer], eSourceType, abyTemp,
+                             eVRTBandDataType);
```

Comment:
The CopyWord() method cannot be applied on the next line, because it relies on compile-time known source type, whereas in below line eVRTBandDataType is a runtime-known data type.